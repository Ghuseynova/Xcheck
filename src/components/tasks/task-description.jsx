import React from 'react';
import 'antd/dist/antd.css';
import PropTypes from 'prop-types';
import { Descriptions } from 'antd';

const TaskDescription = (props) => {
  const task = props.task;
  const { title, author, taskScore, status, description } = task;

  return (
    <>
      <Descriptions column={1} bordered>
        <Descriptions.Item label="Task title">{title}</Descriptions.Item>
        <Descriptions.Item label="Author">{author}</Descriptions.Item>
        <Descriptions.Item label="Max score">{taskScore}</Descriptions.Item>
        <Descriptions.Item label="Status">{status}</Descriptions.Item>
        <Descriptions.Item label="Description">{description}</Descriptions.Item>
        <Descriptions.Item label="Link to this task "></Descriptions.Item>
      </Descriptions>
      <Descriptions title="User Info" column={1} bordered>
        <Descriptions.Item label="Task title">
          Task title Ефеф мидпфлм омфжлм тмжф
        </Descriptions.Item>
        <Descriptions.Item label="Author">ридм\ичвы осмывмо жлмижылвом</Descriptions.Item>
        <Descriptions.Item label="Max score">1000</Descriptions.Item>
      </Descriptions>
    </>
  );
};

TaskDescription.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  taskScore: PropTypes.number.isRequired,
  status: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default TaskDescription;
