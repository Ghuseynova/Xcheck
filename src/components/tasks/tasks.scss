.ant-descriptions-view .ant-descriptions-item-label {
  width: 20%;
}
