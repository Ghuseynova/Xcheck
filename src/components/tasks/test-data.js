const tasks = [
	{
		"taskScore": 100,
		"title": "Task 1",
		"state": "draft",
		"author": "viktorsipach",
		"description": "Task 1 description",
		"items": [
			{
				"id": "task-1-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-bp2",
				"title": "basic_p2",
				"category": "basic",
				"score": "20",
				"description": "basic_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-ep2",
				"title": "extra_p2",
				"category": "extra",
				"score": "50",
				"description": "extra_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-fp2",
				"title": "fines_p2",
				"category": "fines",
				"score": "10",
				"description": "fines_p2 - You need to do some things",
				"mentorCheck": false
			}
		],
		"taskId": "task-1",
		"id": "task-1"
	},
	{
		"taskScore": 100,
		"title": "Task 2",
		"state": "closed",
		"author": "KatiaR",
		"description": "Task 2 description",
		"items": [
			{
				"id": "task-2-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-2-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-2-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			}
		],
		"taskId": "task-2",
		"id": "task-2"
	},
	{
		"taskScore": 100,
		"title": "Task 3",
		"state": "open",
		"author": "YekaterinaKarakulina",
		"description": "Task 3 description",
		"items": [
			{
				"id": "task-3-bp1",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-3-ep1",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-3-fp1",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			}
		],
		"taskId": "task-3",
		"id": "task-3"
	}]

const columns = [
	{
		title: 'Title',
		dataIndex: 'title',
		key: 'title',
	},
	{
		title: 'Author',
		dataIndex: 'author',
		key: 'author',
	},
	{
		title: 'Description',
		dataIndex: 'description',
		key: 'description',
	},
	{
		title: 'Status',
		key: 'state',
		dataIndex: 'state',
	},
	{
		title: 'Action',
		key: 'taskId',
	},
];

const task =
{
	"taskScore": 100,
	"title": "Task 1",
	"state": "draft",
	"author": "viktorsipach",
	"description": "Task 1 description",
	"items": [
		{
			"id": "task-1-bp11",
			"title": "basic_p1",
			"category": "basic",
			"score": "10",
			"description": "basic_p1 - You need to do some things",
			"mentorCheck": true
		},
		{
			"id": "task-1-bp21",
			"title": "basic_p2",
			"category": "basic",
			"score": "20",
			"description": "basic_p2 - You need to do some things",
			"mentorCheck": false
		},
		{
			"id": "task-1-ep11",
			"title": "extra_p1",
			"category": "extra",
			"score": "30",
			"description": "extra_p1 - You need to do some things",
			"mentorCheck": true
		},
		{
			"id": "task-1-ep21",
			"title": "extra_p2",
			"category": "extra",
			"score": "50",
			"description": "extra_p2 - You need to do some things",
			"mentorCheck": false
		},
		{
			"id": "task-1-fp11",
			"title": "fines_p1",
			"category": "fines",
			"score": "10",
			"description": "fines_p1 - You need to do some things",
			"mentorCheck": true
		},
		{
			"id": "task-1-fp21",
			"title": "fines_p2",
			"category": "fines",
			"score": "10",
			"description": "fines_p2 - You need to do some things",
			"mentorCheck": false
		}
	],
	"taskId": "task-11",
	"id": "task-11"
};

export { tasks, columns, task };