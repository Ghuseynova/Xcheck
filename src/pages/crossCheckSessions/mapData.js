const mapData = (databaseData) => {
  const { id, title, author, state, taskId, coefficient, startDate, endDate } = databaseData;
  return {
    key: id,
    id,
    title,
    author,
    state,
    taskId,
    coefficient,
    startDate,
    endDate,
  };
};

export default mapData;
