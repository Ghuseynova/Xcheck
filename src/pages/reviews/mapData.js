const mapData = (databaseData) => {
  const { id, author, state, taskTitle, requestor } = databaseData;
  const grade = databaseData.grade.total;

  return {
    key: id,
    title: id,
    author,
    state,
    taskTitle,
    grade,
    requestor,
  };
};

export default mapData;
