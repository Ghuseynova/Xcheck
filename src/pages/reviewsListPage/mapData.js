const mapData = (databaseData) => {
  const { id, author, state, task } = databaseData;
  return {
    key: id,
    title: id,
    author,
    state,
    task,
  };
};

export default mapData;
