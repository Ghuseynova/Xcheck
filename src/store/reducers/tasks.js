import { POST_TASK_SESSIONS_SUCCESS, POST_TASK_SESSIONS_FAILURE } from '../actions/types';

const tasks = (state = [], action) => {
  switch (action.type) {
    case POST_TASK_SESSIONS_SUCCESS:
      return {
        ...state,
      };

    case POST_TASK_SESSIONS_FAILURE:
      return state;
    default:
      return state;
  }
};

export default tasks;
