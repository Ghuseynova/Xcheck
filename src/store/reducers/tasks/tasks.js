import {
  POST_TASK_SESSIONS_SUCCESS,
  POST_TASK_SESSIONS_FAILURE,
  GET_TASKSTABLE_SESSIONS_SUCCESS,
  GET_TASKSTABLE_SESSIONS_FAILURE,
  GET_TASK_BY_TITLE_SUCCESS,
  GET_TASK_BY_TITLE_FAILURE,
  UPDATE_TASK_SESSION_SUCCESS,
  UPDATE_TASK_SESSION_FAILURE,
} from '../../actions/types/task';

const initialState = {
  formValues: {},
  currentTask: {},
};

const tasks = (state = initialState, action) => {
  switch (action.type) {
    case GET_TASKSTABLE_SESSIONS_SUCCESS:
      return {
        ...state,
        formValues: action.payload,
      };

    case GET_TASKSTABLE_SESSIONS_FAILURE:
      return state;

    case GET_TASK_BY_TITLE_SUCCESS:
      return {
        ...state,
        currentTask: action.payload,
      };
    case GET_TASK_BY_TITLE_FAILURE:
      return {
        ...state,
      };

    case UPDATE_TASK_SESSION_SUCCESS:
      return {
        ...state,
        formValues: {},
      };

    case UPDATE_TASK_SESSION_FAILURE:
      return state;

    case POST_TASK_SESSIONS_SUCCESS:
      return {
        ...state,
      };

    case POST_TASK_SESSIONS_FAILURE:
      return state;

    default:
      return state;
  }
};

export default tasks;
