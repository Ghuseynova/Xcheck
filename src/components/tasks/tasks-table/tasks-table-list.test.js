import React from 'react';
import { shallow } from 'enzyme';
import TasksTable from './tasks-table-list';
import tasks from '../test-data';
import mapData from './map-data-table';

const initialData = tasks[0];
const result = mapData(initialData);

describe('Check mapData', () => {
	it('Check mapData', () => {
		expect(result.author).toEqual(initialData.author);
		expect(result.description).toEqual(initialData.description);
		expect(result.key).toEqual(initialData.taskId);
		expect(result.state).toEqual(initialData.state);
		expect(result.taskId).toEqual(initialData.taskId);
		expect(result.title).toEqual(initialData.title);
	});
});

describe('render ', () => {
	it('render TasksTable', () => {
		const tree = shallow(<TasksTable />);
		expect(tree).toMatchSnapshot();
	});
});



