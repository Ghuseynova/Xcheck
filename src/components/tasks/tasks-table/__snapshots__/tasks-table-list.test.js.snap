// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render  render TasksTable 1`] = `ShallowWrapper {}`;
